### PR TITLE
Me: Use SitesDropdown instead of SelectSite in /me/account

### LINKED ISSUE #451: Me / Site Picker: suggestion: use blog picker styles for primary blog dropdown
Originally raised by @jkudish: it would be nice if the "primary blog" dropdown was similar in functionality to the main blog picker in calypso to allow easier selection of the blog to use.

Feedback from @mtias 

Agreed, whenever we show `Sites` we should use `<Site />` to render them. It would even make sense to put a list of `<Site>`s within `<DropdownSelect />`. 

Design feedback from @folletto 

For reference, here's the mockup of the component for both v1.0 and v1.1:

sitepicker1

sitepicker2

### PR DESCRIPTION
**Pull request by ebinnion**
Fixes #451

Now that we have a `SitesDropdown`, we can retire our use of `SelectSite`. This component swaps out the former for the latter on the `/me/account` route.

__Note:__ There is still some jankiness where switching a site and then refreshing the page may not show the proper primary site. This is because the sites list data is stale after we make the change. So, we'll need to find some way to update the primary site in the sites list.

To test:
- Checkout `fix/me-account-site-dropdown`
- Go to `/me/account`
- Change primary site
- Go to `wordpress.com/me/account`
- Does primary site reflect the change that you just made?

screen shot 2

### CODE REVIEW

**Comment by ockham** [comment]
Now that #1886 has been merged, this is the last occurrence of `me/select-site.jsx`, so maybe you could also remove that file as part of this PR?

**Comment by mtias** [comment]
Nice.

> This is because the sites list data is stale after we make the change. So, we'll need to find some way to update the primary site in the sites list.

This is something we'll properly fix once we migrate site-settings to redux and centralize the data. In the meantime you could just set the primary attribute manually, or change the order of the array. <a href="https://github.com/Automattic/wp-calypso/blob/master/client/lib/sites-list/list.js#L175">This is where we set the attribute</a>.

cc @rralian to keep in mind for settings.

**Comment by ebinnion** [comment]
@mtias - I updated the PR to manually set the primary site. After I did that, I noticed that the old primary site still showed on refresh, which I tracked down to `SitesDropdown` falling back to the primary site if a site is not specified. So, the primary site was being shown until the `/me/settings` API call came back.

I addressed by adding an `isPlaceholder` prop to `SitesDropdown` which, when `true`, will display a `Site` placeholder.

After that, I noticed that the site and chevron could overlap each other. So, I adjusted the styling of the component to use flexbox.

*Overlapping:*
screen shot 5

*Overlapping fixed:*

Chrome:
screen shot 8

screen shot 6


IE11:

screen shot 9

screen shot 7


iPhone4

screen shot 4
 Safari:



@ockham - I went ahead and removed the `select-site` component in this PR as well. :+1:

**Comment by rickybanister** [comment]
Looks good to me.

File: client/me/account/index.jsx [hunk 1]
```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
 		} );
 	},
 
+	onSiteSelect( siteSlug ) {
+		let selectedSite = sites.getSite( siteSlug );
+		if ( sites.getSite( siteSlug ) ) {
```
Comment (dmsnell) [inline client/me/account/index.jsx]: is this a typo? should it be:

```js
if ( selectedSite ) {
```

Comment (ebinnion) [inline client/me/account/index.jsx]: heh... you're right :smile: I created the variable so I didn't have to get the site twice, and then I did it anyways :panda_face:

File: client/me/account/index.jsx [hunk 2]
```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
...
+			this.props.userSettings.updateSetting( 'primary_site_ID', selectedSite.ID );
```
Comment (dmsnell) [inline client/me/account/index.jsx]: Do we not have a Flux-y action to update the primary site? This looks dangerously entangled here.

**Comment by dmsnell** [comment]
@ebinnion looks good. My only real question was about a possible typo and the data path for updating the primary site.

File: client/me/account/index.jsx (hunk 2 above)
Comment (ebinnion) [inline client/me/account/index.jsx]: This is specifically for the `userSettings` module which doesn't have flux. The way we handle data for `user` and `userSettings` could definitely use some work.

**Pull request merged**